lib/llm/src/block_manager/state.rs

```diff
@@ -77,21 +77,32 @@ impl<Metadata: BlockMetadata> KvBlockManagerState<Metadata> {
 
         // Create a NIXL agent if NIXL is enabled and instantiate requested backends
         // TODO: Build a map of NIXL backends to block pools/sets
-        let nixl_agent = match config.runtime.nixl {
+        let nixl_agent = Arc::new(match config.runtime.nixl {
```

Comment:
since we have the full config here, we could look ahead and see if the user is requesting local file, and only if so, create the gds backend.